utils/cache.js

```diff
@@ -89,23 +138,63 @@ const cache = (data = { priority: 2, expiry: CACHE_EXPIRY_TIME_MIN }) => {
 
         res.send = (body) => {
           const cacheValue = {
-            priority: data.priority,
+            priority: priority,
             response: body,
-            expiry: new Date().getTime() + minutesToMilliseconds(data.expiry),
+            expiry: new Date().getTime() + minutesToMilliseconds(expiry),
             size: Buffer.byteLength(body),
           };
           pool.set(key, cacheValue);
+          if (modelKey) {
+            cachedKeys.addCachedKey(modelKey, key);
+          }
           res.send = oldSend;
           return res.send(body);
         };
 
         next();
       }
     } catch (err) {
-      logger.error(`Error while getting cached tasks response ${err}`);
+      logger.error(`Error while getting cached response ${err}`);
+      next();
+    }
+  };
+};
+
+/**
+ * Generate key to save cache.
+ * @param {Object} [request]  HTTP request argument to the middleware function passed by express.
+ * @returns {string} cache key.
+ */
+const generateCacheKey = (request) => {
+  return "__cache__" + request.method + request.originalUrl;
+};
+
+/**
+ * Invalidate the cache for given keys
+ * @param {Object} [options]
+ * @param {Object} [options.invalidationKeys] Array of invalidation keys used in cache middleware.
+ * @returns {function} middleware function to help cache api response.
+ */
+const invalidateCache = (options = {}) => {
```

Comment:
here if the invalidationKeys property is an array. This means that if the property is not an array, the function will try to iterate over it as if it were an array. This will cause an error.

To fix the bug, the function should check if the invalidationKeys property is an array before trying to iterate over it.

Here is an example of how to fix the bug:

const invalidateCache = (options = {}) => {
  const keys = options.invalidationKeys;

  if (!Array.isArray(keys)) {
    throw new Error("Invalidation keys must be an array");
  }

  return async (req, res, next) => {
    try {
      for (const key of keys) {
        const cachedKeysList = cachedKeys.getCachedKeys(key);
        for (const ck of cachedKeysList) {
          pool.evict(ck);
        }
        cachedKeys.removeModelKey(key);
      }
    } catch (err) {
      logger.error(`Error while removing cached response ${err}`);
    } finally {
      next();
    }
  };
};

Reply:
I have added the requested changes